gitlabform/processors/project/tags_processor.py

```diff
@@ -33,13 +36,20 @@ def _process_configuration(self, project_and_group: str, configuration: dict):
                             elif "user_id" in config:
                                 user_ids.add(config["user_id"])
                             elif "user" in config:
-                                user_ids.add(self.gitlab._get_user_id(config["user"]))
+                                users_by_username = self.gl.users.list(
+                                    username=config["user"]
+                                )
+                                if len(users_by_username) == 0:
+                                    raise NotFoundException(
```

Comment:
A minor comment. Is there an appropriate exception from the python-gitlab library that can be caught by the search/list function and raised instead of this one?

Reply:
There wasn't an equivalent error in the [ListMixin](https://python-gitlab.readthedocs.io/en/stable/api/gitlab.html#gitlab.mixins.ListMixin) for "404", so I've wrapped up an empty list result in a `GitlabGetError` which mirrors the exception thrown by `gl.groups.get(config["group"])` when a group is missing.

Reply:
Thanks for the update. Not sure if the behaviour between listing user and group is by design in the python-gitlab library. Maybe @nejch can help clarify.

Reply:
Hey @lhokktyn @amimas! this is expected and normal - you're using two different calls. A `.get()` may return a 404 which will trigger a `GitlabGetError`, for both `users.get(user_id)` and `groups.get(group_id`. 

But a `.list()` may simply return an empty list based on your filters, and this will not raise an exception, for both groups and users.

For this reason I always prefer working with IDs directly instead of performing these searches that need additional API calls but I understand it's more readable for end users to have this as well :+1: 

I'd maybe just use `if not user:` instead of `if len(...)` as per https://peps.python.org/pep-0008/ :sweat_smile:

Reply:
Thank you @nejch for the clarification and your insight.

Reply:
Hi @lhokktyn - has this been resolved already or anything still outstanding? Sorry it's been awhile since we looked at this.

Reply:
Just went through the changes again and I think this is fine. Going to resolve this now.